src/main/java/net/starschema/clouddb/jdbc/BQForwardOnlyResultSet.java

```diff
@@ -126,13 +145,27 @@ public BQForwardOnlyResultSet(
       boolean prefetchedAllRows,
       TableSchema schema,
       @Nullable Long totalBytesProcessed,
-      @Nullable Boolean cacheHit)
+      @Nullable Boolean cacheHit,
+      @Nullable String biEngineMode,
+      @Nullable List<BiEngineReason> biEngineReasons)
       throws SQLException {
     logger.debug("Created forward only resultset TYPE_FORWARD_ONLY");
     this.Statementreference = (Statement) bqStatementRoot;
     this.bigquery = bigquery;
     this.completedJob = completedJob;
     this.projectId = projectId;
+    BiEngineStatistics biEngineStatistics = null;
+    if (completedJob != null) {
+      biEngineStatistics = completedJob.getStatistics().getQuery().getBiEngineStatistics();
+    }
+    if (biEngineStatistics != null) {
```

Comment:
This `if` can be folded into the prior one, since the only way to set `biEngineStatistics` is `if (completedJob != null)`.  Separating them here makes me think that there's another path to this code, so I think it hurts readability.

Reply:
Bump?